### PR TITLE
[AIC-1913] Activate AISimulate CodeRabbit review policy

### PR DESCRIPTION
**Pull request by jasonqinzhou**
## Why and what changed

AISimulate inherited AIC's review guidance under `python/aisimulate/`, but the nested CodeRabbit file was not an active repository-level policy and the repository did not have an explicit fast-before-heavy review sequence.

This PR implements the medium-risk pilot sequence:

`non-draft + review-ready` → `Fast CI + CodeRabbit` → `Codex review` → `exact-SHA Full CI` → `applicable CODEOWNER approval`

The implementation:

- activates a schema-validated root `.coderabbit.yaml` with label-gated review, AISimulate-specific path instructions, linked Dynamo/AIC context, and no automatic approval or request-changes behavior;
- defines the public review contract and risk tiers in `REVIEW.md`;
- adds a reusable Fast CI workflow for repository policy, Python static checks, and Rust formatting;
- retains trusted `pull-request/*` copy branches as an automatic Full CI coverage backstop until the `main` ruleset requires the exact-SHA checks, while preserving an explicit post-review dispatch with a required full `expected_sha`;
- keeps automatic Full CI for trusted `main` and `release/*` lifecycle events;
- puts Cargo policy, Rust workspace tests, application tests, public API builds, release-artifact contracts, and application wheel build/verification/smoke tests in Full CI across amd64 and arm64;
- separates application-wheel verification from trusted release staging, so PR validation never uses the staging environment or token;
- adds an evidence-oriented PR template and CODEOWNER routing for the active policy files; and
- records the adapted AIConfigurator CodeRabbit configuration's immutable source, license, collection hash, modified-overlay hash, and modification notice in both packaged notice files.

## Sequence demonstrated on this PR

1. The branch was synchronized with `main` at `e08f0c3f33bcd725669fbe105085a363eeadd4a9` through signed merge commit `00ad955df951731aa9abdcad6df4b43d9de39c60`.
2. The coverage-backstop update landed in signed commit `c5180f559abf040dcdad3853983f705cfc490d0e`. [Fast CI run 34264016766](https://github.com/ai-dynamo/aisimulate/actions/runs/34264016766) passed on that exact head.
3. CodeRabbit completed a full review of `c5180f5` with no actionable findings. Codex independently reviewed the same commit and found no major issues.
4. Simone Chen approved exact head `c5180f5`; GitHub reports `APPROVED` with all review threads resolved.
5. [Full CI run 34264474398](https://github.com/ai-dynamo/aisimulate/actions/runs/34264474398) required `expected_sha=c5180f559abf040dcdad3853983f705cfc490d0e`, passed the exact-target guard, and completed successfully. Fast CI, Cargo policy, Rust, application tests, public API, release-artifact contracts, and application-wheel verification passed on amd64 and arm64. Trusted staging was skipped by policy.
6. The restored `pull-request/*` trigger is a temporary automatic coverage backstop for copy-pr-bot branches. It does not replace the follow-up repository-settings work to require the Fast and exact-SHA Full CI checks on `main` and require branches to be current.

Pilot note: CodeRabbit activation is automatic from the label. Codex review and the ordered exact-SHA Full CI dispatch were invoked manually on this PR. Making the medium-risk escalation fully automatic requires an approved organization-owned Codex credential/dispatcher; this PR deliberately does not impersonate the PR author or introduce an unapproved secret.

## Review map

- Risk level: medium
- Start with: `.github/workflows/fast-ci.yml`, `.github/workflows/ci.yml`, `REVIEW.md`, and `.coderabbit.yaml`
- Workflow contract changed: yes — PR validation is now split into label-gated Fast CI and exact-SHA Full CI
- Simulator public/serialized behavior changed: no
- Compatibility or rollback concern: CodeRabbit must be installed and authorized for this repository. Linked-repository analysis additionally requires access to those repositories. Keep the `pull-request/*` Full CI backstop until the `main` ruleset requires the exact-SHA checks.

## Evidence

- Final tested head: `c5180f559abf040dcdad3853983f705cfc490d0e`
- Current-main synchronization: signed merge commit `00ad955df951731aa9abdcad6df4b43d9de39c60`; PR base is `e08f0c3f33bcd725669fbe105085a363eeadd4a9`
- CodeRabbit full exact-head review: no actionable findings
- Codex exact-head review: no major issues
- Exact-head CODEOWNER review: approved by Simone Chen
- Full CI: [run 34264474398](https://github.com/ai-dynamo/aisimulate/actions/runs/34264474398) succeeded; exact-target guard, Fast CI, Cargo policy, Rust, application, public API, release-artifact, and application-wheel jobs passed on both architectures
- Focused policy/source-compliance pytest suite: 101 passed
- Generated CODEOWNERS strict coverage: 2,837 / 2,837 tracked files, zero catch-all-only files
- `python scripts/check_copyright.py`: 1,063 tracked source files passed
- `actionlint` v1.7.7 on both workflows: passed
- Current CodeRabbit YAML 1.2 schema validation: passed
- Root and packaged `THIRD_PARTY_NOTICES.md`: byte-identical
- Ruff, compileall, `cargo fmt --all -- --check`, generated-artifact diff, and `git diff --check`: passed

CodeRabbit's aggregate advisory retained a Cross-Layer Contract warning about `py_ops.rs`, `use_qk_norm`, and a schema-version change. Those changes came from the synchronized `main` history rather than this PR's policy delta. The policy update itself introduces no simulator formula, public schema, or predicted-output change.

## Modeling or data provenance

No simulator formula, selection logic, public input, serialized field, performance value, or predicted output changed. The checked-in MoE oracle samples are byte-for-byte unchanged except for `_source` metadata naming the internal `_engine_query` surface actually used by the generator.

The adapted CodeRabbit policy came from `ai-dynamo/aiconfigurator` commit `13b5cf2697876692b0a52098266c81162add11fc`, path `.coderabbit.yaml`, under Apache-2.0. The collected upstream file hash is `5fb7a61a53f71f476169fa8e2419d3073c8d7e96206d8986b7d4fbb0f11fbcdd`; the modified AISimulate overlay hash is `70960b92994caaad52f806bd5c618353670c9e754a4d61525962f561360d9d48`.

## Tracking

- Primary: [AIC-1913 — Activate and configure CodeRabbit for AISimulate](https://linear.app/nvidia/issue/AIC-1913/review-activate-and-configure-coderabbit-for-aisimulate)
- Related: [AIC-1910 — Codify the deep AISimulate reviewer contract](https://linear.app/nvidia/issue/AIC-1910/review-codify-the-deep-aisimulate-reviewer-contract-and-knowledge)
- Related: [AIC-1914 — Create the review-ready intake and escalation workflow](https://linear.app/nvidia/issue/AIC-1914/process-create-the-review-ready-intake-risk-tiers-and-escalation)

### CODE REVIEW

**Comment by jasonqinzhou** [comment]
@coderabbitai configuration

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@codex review

File: python/aisimulate/.claude/rules/rust-core/parity.md [hunk 1]
```diff
@@ -53,41 +56,46 @@ estimation math. Concretely:
...
-  removed together in the deprecation-cleanup PR. New per-op access goes
-  through `EngineHandle.evaluate_ops_json` / `evaluate_ops_sol_json`, the
-  per-phase surface (`run_static_per_op`), or whole runs. `Operation.query`
-  overrides are limited to the orchestration whitelist (AFD comm ops, the
-  deprecated `Mamba2` composite) whose bodies compose ENGINE-evaluated twin
-  ops.
+- The deprecated `PerfDatabase.query_*` and public `Operation.query` shims
+  have been removed. Do not reintroduce them. New per-op access goes through
+  `EngineHandle.evaluate_ops_json` / `evaluate_ops_sol_json`, the per-phase
+  surface (`run_static_per_op`), or whole runs. `Operation.query` bodies are
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply python/aisimulate/.claude/rules/rust-core/parity.md:66]: Fixed in bdb1c35 after verifying the generator behavior. The docstring and checked-in fixture now name `_engine_query`, and the regeneration instructions explicitly use `python/aisimulate` as the working directory.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Evidence
+
+<!--
+List exact commands and results. Identify the tested commit when evidence came
+from CI.
+-->
+
+- Tests:
+- Negative or boundary cases:
+- Before/after output, trace, benchmark, or golden diff:
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/pull_request_template.md:37]: Fixed in bdb1c35. The Evidence section now asks authors to state the independent expected-value derivation for exact numerical assertions.

**Comment by jasonqinzhou** [comment]
@codex review

**Comment by jasonqinzhou** [comment]
@codex review

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@codex review

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@codex review

**Comment by jasonqinzhou** [comment]
@coderabbitai review

File: .coderabbit.yaml [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+  auto_review:
+    # Keep the pilot opt-in: applying the review-ready label triggers review
+    # even while automatic review is disabled globally.
+    enabled: false
+    drafts: false
+    auto_incremental_review: true
+    auto_pause_after_reviewed_commits: 5
+    base_branches:
+      - "release/.*"
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .coderabbit.yaml:47]: No code change: this is a false positive. CodeRabbit documents `base_branches` as additional branches other than the default branch, so `main` remains eligible. This PR targets `main`, carries `review-ready`, and CodeRabbit used the root configuration successfully. Reference: https://docs.coderabbit.ai/reference/configuration

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,27 +19,15 @@ permissions:
...
-          python-version: "3.12"
-      - name: Check NVIDIA/Apache-2.0 SPDX headers
-        run: python scripts/check_copyright.py
-      - name: Check packaged legal-file copies
-        run: python scripts/check_packaged_legal_files.py
+  fast-ci:
+    name: Fast CI
+    uses: ./.github/workflows/fast-ci.yml
+    with:
+      expected_sha: ${{ inputs.expected_sha || github.sha }}
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/ci.yml]: Fixed in 36cd53c. Full CI now has a `Verify Exact Target` job that rejects `expected_sha` unless it equals the immutable `github.sha`; Fast CI receives that same `github.sha`, and every heavyweight checkout uses the same workflow SHA. The repository policy test pins this contract.

File: crates/core/parity_tests/perfmodel/gen_op_oracle.py [hunk 5]
```diff
@@ -409,7 +409,10 @@ def main() -> None:
 
     header = {
         "_regenerate": (".venv/bin/python aic-core/rust/aiconfigurator-core/parity_tests/gen_op_oracle.py"),
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply crates/core/parity_tests/perfmodel/gen_op_oracle.py:411]: No code change here. The recorded `aic-core/rust/aiconfigurator-core/parity_tests` path is a tracked compatibility symlink to `crates/core/parity_tests/perfmodel`, so it does resolve to the current generator. I tested the proposed direct-path rewrite in a fresh locked environment; regeneration changed 10 latency samples, including material DeepEP values, without modeling rationale. Those numerical changes were restored rather than hidden in this CI-policy PR; any oracle drift needs a separate modeling review with explained before/after evidence.

File: REVIEW.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+Do not report preference-only naming, formatting, or documentation nits already
+covered by automated tools. Do not treat bot approval, `MERGEABLE`, CODEOWNERS,
+DCO, or a small green check set as merge readiness. Technical findings and
+governance status are separate; merge readiness is assessed on the exact head.
+
+## Risk-tiered review and CI
+
+The `review-ready` label is the explicit admission gate for the inexpensive
+evidence-gathering stage. Once it is applied, Fast CI and CodeRabbit run in
+parallel for every non-draft PR. Medium- and high-risk PRs also require a Codex
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply REVIEW.md:31]: Verified, but no workflow change is appropriate. The review-ready label is explicitly the admission gate for direct pull_request Fast CI runs. Full CI workflow_dispatch is a separate maintainer-controlled admission after current-head bot review, as REVIEW.md lines 44-52 state; main/release/pull-request branch pushes are trusted lifecycle events rather than untrusted PR admission events. Requiring PR label context inside workflow_call would incorrectly block those non-PR callers, which have no pull_request payload.

File: .coderabbit.yaml [hunk 7]
```diff
@@ -0,0 +1,359 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# yaml-language-server: $schema=https://coderabbit.ai/integrations/schema.v2.json
+# Active repository-level CodeRabbit policy. The copy under
+# python/aisimulate/ is preserved migration provenance and is not active here.
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .coderabbit.yaml:10]: Fixed in 6a6d223. The active root policy now records the upstream repository, immutable commit, original path, Apache-2.0 license, preserved 2025-2026 copyright, and AISimulate modification. Matching AIConfigurator entries were added to both byte-identical THIRD_PARTY_NOTICES.md copies.

File: .github/codeowners/test_aisimulate_policy.py [hunk 8]
```diff
@@ -168,3 +171,33 @@ def test_dependency_policy_covers_every_rust_manifest_root() -> None:
...
+        "Release Artifact Contract",
+        "Application Wheel",
+    ):
+        assert expensive_gate in full
+
+    assert "uses: ./.github/workflows/fast-ci.yml" in full
+    assert "needs: fast-ci" in full
+    assert 'EXPECTED_SHA: ${{ inputs.expected_sha }}' in full
+    assert 'RUN_SHA: ${{ github.sha }}' in full
+    assert 'expected_sha: ${{ github.sha }}' in full
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/codeowners/test_aisimulate_policy.py:207]: Fixed in 6a6d223: the policy test now asserts both the fast-ci dependency on verify-target and the active SHA mismatch guard.

**Comment by jasonqinzhou** [comment]
@coderabbitai review current head 6a6d2239639ac002e50b1c862babc9934df7d9d8.

**Comment by jasonqinzhou** [comment]
@codex review current head 6a6d2239639ac002e50b1c862babc9934df7d9d8 for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai review current head 14d236199a0fc3c4bfe375ebf64272bebde0b69a. The prior P1 is a false positive: the existing auto_review.labels block already contains review-ready (plus negative filters); enabled:false with labels is CodeRabbit documented label-driven opt-in, and the default branch is always included. Commit 14d2361 adds a YAML-parsing regression test for that exact contract.

**Comment by jasonqinzhou** [comment]
@codex review current head 14d236199a0fc3c4bfe375ebf64272bebde0b69a for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai review current head 6a1142e8ff5abd37c1c4a03c51658e9462de00ba. The stale Python oracle references are now aligned to _engine_query; the golden samples are unchanged.

**Comment by jasonqinzhou** [comment]
@codex review current head 6a1142e8ff5abd37c1c4a03c51658e9462de00ba for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai review current head 6a1142e8ff5abd37c1c4a03c51658e9462de00ba. Please review the final two documentation-only oracle-reference changes.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review current head 6a1142e8ff5abd37c1c4a03c51658e9462de00ba. Incremental review could not access the parent commit, so please review the complete current PR.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

File: .github/workflows/fast-ci.yml [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+  group: fast-ci-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  policy:
+    name: Repository Policy
+    if: >-
+      github.event_name != 'pull_request' ||
+      (!github.event.pull_request.draft &&
+       contains(github.event.pull_request.labels.*.name, 'review-ready'))
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/fast-ci.yml:27]: Fixed in 945e882. pull-request/* no longer automatically triggers Full CI; PR validation is maintainer-dispatched after current-head Fast CI and required bot reviews. main and release/* retain lifecycle push runs, and a parsed-workflow regression test locks the trigger boundary.

**Comment by jasonqinzhou** [comment]
@codex review current head 945e88264cbcd213e21c40c086e14a6f5d785f7c for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@coderabbitai resume

**Comment by jasonqinzhou** [comment]
@coderabbitai review

File: .github/codeowners/test_aisimulate_policy.py [hunk 10]
```diff
@@ -168,3 +173,45 @@ def test_dependency_policy_covers_every_rust_manifest_root() -> None:
...
+    assert 'if [[ -n "${EXPECTED_SHA}" && "${EXPECTED_SHA}" != "${RUN_SHA}" ]]; then' in full
+
+
+def test_coderabbit_is_opted_in_by_review_ready_label() -> None:
+    policy = yaml.safe_load((ROOT / ".coderabbit.yaml").read_text())
+    auto_review = policy["reviews"]["auto_review"]
+
+    assert auto_review["enabled"] is False
+    assert auto_review["labels"] == ["review-ready", "!wip", "!do-not-review"]
+    assert auto_review["drafts"] is False
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/codeowners/test_aisimulate_policy.py:246]: Fixed in b0c307a. The parsed policy regression test now locks request_changes_workflow, base_branches, and all ignored title keywords in addition to the existing label, enabled, and draft controls.

File: THIRD_PARTY_NOTICES.md [hunk 11]
```diff
@@ -13,6 +13,22 @@ distributed with those packages.
 Unless otherwise stated, AISimulate file paths in this document are relative
 to `python/aisimulate/` in the repository source tree.
 
+## AIConfigurator
+
+The repository-root `.coderabbit.yaml` is adapted and modified from
+AIConfigurator's `.coderabbit.yaml` at commit
+`13b5cf2697876692b0a52098266c81162add11fc`. The original imported policy is
+preserved at `python/aisimulate/.coderabbit.yaml` as migration provenance.
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply THIRD_PARTY_NOTICES.md:21]: Fixed in b0c307a. Both byte-identical notice files now record the pinned upstream/preserved-file SHA-256 and the modified root overlay SHA-256; both digests were reverified locally.

File: .github/workflows/ci.yml [hunk 12]
```diff
@@ -1,39 +1,49 @@
...
+        type: string
+        default: ""
   push:
-    # Velonix admits canonical pushes, not untrusted pull_request events.
+    # PR validation is admitted through workflow_dispatch after review. Keep
+    # automatic runs only for post-merge and release lifecycle validation.
     branches:
       - main
-      - "pull-request/*"
       - "release/*"
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/ci.yml:17]: Fixed in b0c307a. The application-wheel matrix now always builds, verifies, and smoke-tests wheels, including workflow_dispatch. A separate stage-application-wheel matrix runs only for trusted main/release pushes, downloads the pinned artifact, and alone acquires the automated-release environment and secrets.

**Comment by jasonqinzhou** [comment]
@codex review current head b0c307a9c9c25fc0ab4a76cb3ba6670f4c0f02f7 for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

File: crates/core/parity_tests/perfmodel/gen_op_oracle.py [hunk 13]
```diff
@@ -23,7 +23,7 @@
 tests read the two ``op`` slices of that one file, rebuild the same ops
 against the same shipped parquet files and assert a relative error <= 1e-9.
 
-Regenerate (from the repo root, after `git lfs pull`):
+Regenerate (from ``python/aisimulate``, after ``git lfs pull``):
 
     .venv/bin/python aic-core/rust/aiconfigurator-core/parity_tests/gen_op_oracle.py
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply crates/core/parity_tests/perfmodel/gen_op_oracle.py:28]: Verified false positive; no code change. The documented path exists as a tracked symlink to crates/core/parity_tests/perfmodel. Running runpy on that exact path from python/aisimulate resolves REPO_ROOT to python/aisimulate/aic-core, whose tracked src symlink points to ../src, and os.path.normpath(OUT_PATH) reaches the tracked aiconfigurator-core/src symlink to crates/core/src/perfmodel. The documented command was also executed earlier: imports succeeded and generation ran; it exposed unrelated numerical drift, so the golden refresh was intentionally reverted per REVIEW.md.

**Comment by jasonqinzhou** [comment]
@coderabbitai full review

File: .github/codeowners/test_aisimulate_policy.py [hunk 14]
```diff
@@ -168,3 +173,64 @@ def test_dependency_policy_covers_every_rust_manifest_root() -> None:
...
+    assert full_config["on"]["push"]["branches"] == ["main", "release/*"]
+    application_wheel = full_config["jobs"]["application-wheel"]
+    assert "if" not in application_wheel
+    assert any(
+        step.get("name") == "Verify exact staged wheel"
+        for step in application_wheel["steps"]
+    )
+    assert "github.event_name == 'push'" in full_config["jobs"][
+        "stage-application-wheel"
+    ]["if"]
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/codeowners/test_aisimulate_policy.py]: Fixed in 8e23928. The parsed-workflow test now requires exactly one verifier step, its exact executable command, no conditional skip, no continue-on-error, and the exact push-only main/release staging predicate.

**Comment by jasonqinzhou** [comment]
@codex review current head 8e2392822d35e0ef8a50951f46973c430339c582 for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@coderabbitai review

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -1,39 +1,49 @@
...
+name: Full CI
 
 on:
   workflow_dispatch:
+    inputs:
+      expected_sha:
+        description: Optional exact commit expected for this run
+        required: false
+        type: string
+        default: ""
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply .github/workflows/ci.yml]: Fixed in 9d7c681. workflow_dispatch.expected_sha is now required with no default; REVIEW.md requires the reviewed full SHA, and the parsed-workflow regression test locks both properties. Push lifecycle runs still use immutable github.sha.

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Comment by jasonqinzhou** [comment]
@codex review current head 9d7c68112d2c49ffc0d9fcc57e2cd542aefa16a8 for the medium-risk pilot.

**Comment by jasonqinzhou** [comment]
@coderabbitai review

**Review by Arsene12358 — APPROVED** [review]
Approving. I re-verified the PR's evidence independently rather than taking the description at face value: both cited runs are real and ran on `9d7c68112d2c49ffc0d9fcc57e2cd542aefa16a8` (Fast CI 33923440930 on `pull_request`, Full CI 33924334424 on `workflow_dispatch`, all 16 heavy jobs green on amd64 and arm64); the five new action pins resolve to their tags; the upstream collection hash `5fb7a61a…` and overlay hash `70960b92…` match; the root config validates against `schema.v2.json`, and the `enabled: false` + `labels: [review-ready, …]` opt-in is exactly the semantics the schema documents (so the earlier `base_branches` P1 was indeed a false positive); and every Fast CI job reproduces green locally on a merge with current `main` (101 policy tests, CODEOWNERS in sync, `check_copyright`, `check_packaged_legal_files`, ruff, compileall, `git diff --check`, actionlint all clean). The `_regenerate` symlink rebuttal is also correct (`aic-core/rust/aiconfigurator-core/{parity_tests,src}` are tracked symlinks). No blockers.

One request that needs a repository-settings change rather than a code change, ideally landed together with this PR:

**Make the exact-SHA Full CI a required status check on `main`.** Once `pull-request/*` leaves the push trigger, a PR's only route to the Rust, application, cargo-deny, public-API and release-contract jobs is a maintainer dispatching Full CI by hand. The ruleset on `main` today requires one approval, code-owner review and thread resolution, but has no `required_status_checks` rule (`GET /repos/ai-dynamo/aisimulate/rules/branches/main`; GraphQL `isRequired` is `false` for every check on this PR and on #120). So the REVIEW.md rule that "all required conversations, approvals, and exact-head checks must be complete before merge" is procedural only: a CODEOWNER approval alone merges a head that never ran Full CI. On `main` today the copy-pr-bot copies run the full suite automatically (for example `pull-request/113` and `pull-request/119`, both fork PRs whose copy heads equal the PR heads), so this is a real reduction in automatic coverage unless the ruleset backs it.

Dispatched check runs already attach to the PR head SHA (this PR shows all of them in its checks tab), so adding them as required checks makes the manual dispatch mandatory with no workflow change:

```
Verify Exact Target
Rust (amd64)                          Rust (arm64)
Application Tests (amd64)             Application Tests (arm64)
Cargo Deny (workspace, amd64)         Cargo Deny (workspace, arm64)
Public API Rust (amd64)               Public API Rust (arm64)
Release Artifact Contract (amd64)     Release Artifact Contract (arm64)
Application Wheel (amd64)             Application Wheel (arm64)
```

plus the three direct Fast CI job names (`Repository Policy`, `Python Static Checks`, `Rust Format`) and "require branches to be up to date", so a head that goes stale against `main` has to be re-dispatched. If the ruleset change cannot land alongside this PR, I would keep the `pull-request/*` push trigger until it does, and note in REVIEW.md that the Full CI gate is currently procedural.

**Review by tedzhouhk — APPROVED** [review]

**Comment by jasonqinzhou** [comment]
@coderabbitai review current head c5180f559abf040dcdad3853983f705cfc490d0e. This update syncs current main and restores the pull-request/* Full CI coverage backstop.

**Comment by jasonqinzhou** [comment]
@codex review current head c5180f559abf040dcdad3853983f705cfc490d0e for the medium-risk pilot. Focus on the restored pull-request/* Full CI backstop and its credential/staging boundary.

**Review by simone-chen — APPROVED** [review]
